ui/apps/platform/src/Containers/Dashboard/PatternFly/hooks/useRestQuery.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react';
+import { CancellableRequest } from 'services/cancellationUtils';
+
+export type UseRestQueryReturn<ReturnType> = {
+    data: ReturnType | undefined;
+    loading: boolean;
+    error: Error | null;
+};
+
+export default function useRestQuery<ReturnType>(
+    cancellableRequestFn: () => CancellableRequest<ReturnType>
+): UseRestQueryReturn<ReturnType> {
+    const [data, setData] = useState<ReturnType>();
+    const [loading, setLoading] = useState<boolean>(true);
+    const [error, setError] = useState<Error | null>(null);
+
+    useEffect(() => {
+        const { request, cancel } = cancellableRequestFn();
+
+        setError(null);
+
+        request
+            .then((groups) => {
```

Comment:
Is `groups` a left over from the original specific hook?

Because `data` is in function scope, I am interested in your thought of a generic argument name, because I have struggled with an intuitive naming convention this situation.

Reply:
Aha, I figured I would miss one!

I like `result` to describe the result of the request, and `data` to represent the result stored client side. I'll push a fix up.